### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-scrollable

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1485]
## Description
This task describes the Storybook implementation for the sd-scrollable component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-scrollable can be found here: [sd-scrollable Notes](https://www.notion.so/matuzo/sd-scrollable-10cf3f6e1bc080e6af27d02209dd3bcf).

--------------------

## Comment
**Problem:** If the content doesn't contain interactive elements, it becomes inaccessible to keyboard users in some browsers (e.g. Chromium-based browsers) because scrollable areas are not keyboard accessible by default. (Note: that may change in Chrome 130+, but you'd still have to test it with different browsers).
**Solution:** You could turn the content area into a scrollable region using role=region or role=group, tabindex=0, aria-label, or aria-labelledby.

**Problem:** The buttons have no accessible name.
**Solution:** Add a label to the icons.

**Problem:** Focus is lost on the scroll buttons when you reach the end of the scrollable area because the buttons are removed from the DOM.
**Solution:** Move focus to the other active button when you reach the end of the area.

**Problem:** There's no announcement when screen reader users press the scroll buttons.
**Solution:** Use a live region to announce something like "scrolled up/down" or "scrolled to the beginning/end".

## Best Practices
- If possible, avoid forcing scrollable content.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#keyboard
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value
- https://www.w3.org/WAI/WCAG22/quickref/#status-messages

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
